pkg/policy/rule.go

```diff
@@ -263,25 +264,35 @@ func (r *rule) resolveL4Policy(ctx *SearchContext, state *traceState, result *L4
 		if len(r.Ingress) == 0 {
 			ctx.PolicyTrace("    No L4 rules\n")
 		}
-		for _, r := range r.Ingress {
-			cnt, err := mergeL4(ctx, "Ingress", r.FromEndpoints, r.ToPorts, result.Ingress)
+		result.Ingress.SourceRuleLabels = make(labels.Labels)
```

Comment:
I've a feeling `result` is being used multiple times by the caller of `resolveL4Policy`, aren't you going to overwrite the `SourceRuleLabels` on every loop?

https://github.com/cilium/cilium/blob/5fc5b37a2b9d31da0bef6ea61eaac63f7fdf24ff/pkg/policy/repository.go#L144-L154

Reply:
You are correct. In fact, the looping might include all endpoints so this is the wrong place to accumulate this information :(